Build and publish pydeephaven-ticking wheels (#5290)
deephaven/deephaven-core: RELEASE.md

It is released to [GitHub releases](https://github.com/deephaven/deephaven-core/releases) in the [syft](https://github.com/anchore/syft) json format.

### Deephaven python client

> The Deephaven python client is released as the `pydeephaven` wheel at [PyPi](https://pypi.org/project/pydeephaven/) and
> `pydeephaven-ticking` wheels at [PyPi](https://pypi.org/project/pydeephaven-ticking/).

Reviewer: Seems strangely worded, where the link is on PyPi both times. Suggest switching it around:
```suggestion
The Deephaven python client wheels are released on PyPi as [`pydeephaven`](https://pypi.org/project/pydeephaven/) and
[`pydeephaven-ticking`](https://pypi.org/project/pydeephaven-ticking/).
```

The "Publish pydeephaven to PyPi" uploads the pydeephaven wheel to [PyPi](https://pypi.org/project/pydeephaven/).
If this step fails, the pydeephaven wheel from the "Upload Artifacts" step can be uploaded manually.

> There is no automatic uploads for the pydeephaven-ticking wheels to [PyPi](https://pypi.org/project/pydeephaven-ticking/).

Reviewer: Isn't that the `Publish pydeephaven-ticking to PyPi` action that you're adding with this PR? Not clear why pydeephaven-ticking isn't automatic but the others are based on this comment.

Author: Forgot to update, will do.
